src/asmcnc/core_UI/popups.py

```diff
@@ -0,0 +1,280 @@
+        self.main_layout_spacing = main_layout_spacing
+        self.popup_type = popup_type
+        self.popup_image = popup_image
+        self.popup_image_size_hint = popup_image_size_hint
+        self.button_one_text = self.l.get_str(button_one_text)
+        self.button_one_callback = button_one_callback
+        self.button_two_text = (
+            self.l.get_str(button_two_text) if button_two_text is not None else None
+        )
+        self.button_two_callback = button_two_callback
+        self.button_one_background_color = button_one_background_color
+        self.button_two_background_color = button_two_background_color
+        self.button_one_background_normal = (
+            self.button_one_background_normal
+            if button_one_background_color is None
+            else ""
+        )
+        self.button_two_background_normal = (
+            self.button_two_background_normal
+            if button_two_background_color is None
+            else ""
+        )
+        if self.popup_type is not None:
+            if self.popup_type.value is not None:
+                self.separator_color = self.popup_type.value["separator_color"]
+
+        self.build()
+
+    def build(self):
+        text_size_x = dp(
+            utils.get_scaled_width(
+                self.popup_width - utils.get_scaled_width(self.main_label_size_delta)
+            )
+        )
+
+        self.main_label = Label(
+            size_hint_y=2,
+            text_size=(text_size_x, None),
+            halign="center",
+            valign="middle",
+            text=self.l.get_str(self.main_string),
+            color=(0, 0, 0, 1),
+            padding=utils.get_scaled_tuple(self.main_label_padding),
+            markup=True,
+            font_size=str(utils.get_scaled_width(14)) + "sp",
+        )
+
+        self.button_layout = self.build_button_layout()
+
+        self.main_layout = BoxLayout(
+            orientation="vertical",
+            spacing=utils.get_scaled_tuple(
+                self.main_layout_spacing, orientation="vertical"
+            ),
+            padding=utils.get_scaled_tuple(self.main_layout_padding),
+        )
+
+        image = self.get_image()
+        if image is not None:
+            self.main_layout.add_widget(image)
+
+        self.main_layout.add_widget(self.main_label)
+        self.main_layout.add_widget(self.button_layout)
+
+        self.content = self.main_layout
+
+    def build_button_layout(self):
+        button_layout = BoxLayout(
+            orientation="horizontal",
+            spacing=utils.get_scaled_tuple(
+                self.button_layout_spacing, orientation="horizontal"
+            ),
+            padding=utils.get_scaled_tuple(self.button_layout_padding),
+        )
+
+        for button in self.build_buttons():
+            button_layout.add_widget(button)
+        return button_layout
+
+    def on_button_pressed(self, callback):
+        self.dismiss()
+        callback()
```

Comment:
have callback() before dismiss() - better that the popup is unresponsive for a microsecond or doesn't disappear before the next thing happens, then getting a slightly unresponsive screen that then does something else